name: mobile_app
description: "ASIS Mobile App"
publish_to: 'none'
version: 1.0.0

environment:
  sdk: 3.5.1
  flutter: 3.24.1

dependencies:
  core_analytics_firebase:
    path: ../../core/analytics_firebase
  core_authenticator:
    path: ../../core/authenticator
  core_common:
    path: ../../core/common
  core_database_isar:
    path: ../../core/database_isar
  core_datastore:
    path: ../../core/datastore
  core_designsystem:
    path: ../../core/designsystem
  core_domain:
    path: ../../core/domain
  core_model:
    path: ../../core/model
  core_network:
    path: ../../core/network
  core_ui:
    path: ../../core/ui
  cupertino_icons: 1.0.8
  dynamic_color: 1.7.0
  envied: 0.5.4+1
  feature_auth:
    path: ../../feature/auth
  feature_feed:
    path: ../../feature/feed
  feature_home:
    path: ../../feature/home
  feature_quest:
    path: ../../feature/quest
  feature_settings:
    path: ../../feature/settings
  firebase_analytics: 11.3.0
  firebase_auth: 5.2.0
  firebase_core: 3.4.0
  firebase_remote_config: 5.1.0
  flutter:
    sdk: flutter
  flutter_hooks: 0.20.5
  flutter_localizations:
    sdk: flutter
  go_router: 14.2.3
  hooks_riverpod: 2.5.1
  intl: 0.19.0
  isar:
    hosted: https://pub.isar-community.dev/
    version: 4.0.3
  isar_flutter_libs:
    hosted: https://pub.isar-community.dev/
    version: 4.0.3
  package_info_plus: 8.0.2
  path_provider: 2.1.4
  riverpod_annotation: 2.3.5
  shared_preferences: 2.3.1
  url_launcher: 6.3.0

dependency_overrides:
  # workaround
  # So, because feature_quest depends on both flutter_test from sdk and riverpod_generator 2.4.2, version solving failed.
  meta: 1.15.0
  # workaround
  # https://stackoverflow.com/questions/77947590/error-the-type-jsobject-cant-be-used-as-supertype-while-deploying-flutter
  web: 0.4.2

dev_dependencies:
  build_runner: 2.4.12
  envied_generator: 0.5.4+1
  go_router_builder: 2.7.1
  riverpod_generator: 2.4.2
  yumemi_lints: 2.1.0

flutter:
  uses-material-design: true
